runtime/runtime_controller.h

```diff
@@ -22,13 +22,89 @@
+  ///                                         usually the `Engine` instance.
+  /// @param      vm                          A reference to a running Dart VM.
+  ///                                         The runtime controller must be
+  ///                                         collected before the VM is
+  ///                                         destroyed (this order is
+  ///                                         guaranteed by the shell).
+  /// @param[in]  isolate_snapshot            The isolate snapshot used to start
+  ///                                         the root isolate managed by this
+  ///                                         runtime controller. The isolate
+  ///                                         must be transitioned into the
+  ///                                         running phase manually by the
+  ///                                         caller.
+  /// @param[in]  task_runners                The task runners used by the shell
+  ///                                         hosting this runtime controller.
+  ///                                         This may be used by the isolate to
+  ///                                         scheduled asynchronous texture
+  ///                                         uploads or post tasks to the
+  ///                                         platform task runner.
+  /// @param[in]  snapshot_delegate           The snapshot delegate used by the
+  ///                                         isolate to gather raster snapshots
+  ///                                         of Flutter view hierarchies.
+  /// @param[in]  io_manager                  The IO manager used by the isolate
+  ///                                         for asynchronous texture uploads.
+  /// @param[in]  unref_queue                 The unref queue used by the
+  ///                                         isolate to collect resources that
+  ///                                         may reference resources on the
+  ///                                         GPU.
+  /// @param[in]  image_decoder               The image decoder
+  /// @param[in]  advisory_script_uri         The advisory script URI (only used
+  ///                                         for debugging). This does not
+  ///                                         affect the code being run in the
+  ///                                         isolate in any way.
+  /// @param[in]  advisory_script_entrypoint  The advisory script entrypoint
+  ///                                         (only used for debugging). This
+  ///                                         does not affect the code being run
+  ///                                         in the isolate in any way. The
+  ///                                         isolate must be transitioned to
+  ///                                         the running state explicitly by
+  ///                                         the caller.
```

Comment:
My understanding at this point is that these don't actually do anything at all, and don't show up anywhere meaningfully for debuggers.  Is that correct?

Reply:
I think these still show up when you use the debugger from the observatory. It might be a good idea to just remove these from the engine and see what breaks really.